File: lab-04/app/src/main/java/ute/ltm/ltm_bt04/MainActivity.java
package ute.ltm.ltm_bt04;

import ute.ltm.ltm_bt04.ui.SQLiteActivity;
import ute.ltm.ltm_bt04.ui.SharedPrefActivity;
import ute.ltm.ltm_bt04.ui.UserListActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;


public class MainActivity extends AppCompatActivity {

    Button btnGoToSQLite;
    Button btnGoToDataBinding;
    Button btnGoToSharedPref;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnGoToSQLite = findViewById(R.id.btnGoToSQLite);
        btnGoToSQLite.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SQLiteActivity.class);
                startActivity(intent);
            }
        });

        btnGoToDataBinding = findViewById(R.id.btnGoToDataBinding);
        btnGoToDataBinding.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, UserListActivity.class);
                startActivity(intent);
            }
        });

        btnGoToSharedPref = findViewById(R.id.btnGoToSharedPref);
        btnGoToSharedPref.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SharedPrefActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: lab-10/app/src/main/java/ute/android/lab_10/MainActivity.kt
package ute.android.lab_10

import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import ute.android.lab_10.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var dbHelper: DatabaseHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        dbHelper = DatabaseHelper(this)

        binding.btnLogin.setOnClickListener {
            val user = binding.edtUsername.text.toString()
            val pass = binding.edtPassword.text.toString()

            if (dbHelper.checkUser(user, pass)) {
                Toast.makeText(this, "Đăng nhập thành công!", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "Sai tài khoản hoặc mật khẩu", Toast.LENGTH_SHORT).show()
            }
        }

        binding.btnRegister.setOnClickListener {
            val user = binding.edtUsername.text.toString()
            val pass = binding.edtPassword.text.toString()
            dbHelper.addUser(User(username = user, password = pass))
            Toast.makeText(this, "Đăng ký xong rồi đó bro!", Toast.LENGTH_SHORT).show()
        }
    }
}